In [1]:
import pandas as pd

In [2]:
from ydata_profiling import ProfileReport

In [3]:
df=pd.read_csv('Tunisair_flights_dataset.csv')

In [4]:
df.head(10)

Filght_date Flight_ID Departure point Arrival point  \
0  2016-01-03  TU 0712              CMN           TUN   
1  2016-01-13  TU 0757              MXP           TUN   
2  2016-01-16  TU 0214              TUN           IST   
3  2016-01-17  TU 0480              DJE           NTE   
4  2016-01-17  TU 0338              TUN           ALG   
5  2016-01-17  TU 0283              TLS           TUN   
6  2016-01-18  TU 0514              TUN           BCN   
7  2016-01-18  TU 0716              TUN           ORY   
8  2016-01-18  TU 0752              TUN           FCO   
9  2016-01-18  TU 0996              TUN           NCE   

  Scheduled_departure_time Scheduled_arrival_time STATUS Aircraft_code  \
0      2016-01-03 10:30:00    2016-01-03 12.55.00    ATA     TU 32AIMN   
1      2016-01-13 15:05:00    2016-01-13 16.55.00    ATA     TU 31BIMO   
2      2016-01-16 04:10:00    2016-01-16 06.45.00    ATA     TU 32AIMN   
3      2016-01-17 14:10:00    2016-01-17 17.00.00    ATA     TU 736IOK   
4      2016-01-17 14:30:00    2016-01-17 15.50.00    ATA     TU 320IMU   
5      2016-01-17 16:20:00    2016-01-17 18.15.00    ATA     TU 736IOP   
6      2016-01-18 07:15:00    2016-01-18 09.00.00    ATA     TU 32AIMH   
7      2016-01-18 07:35:00    2016-01-18 09.55.00    ATA     TU 32AIMI   
8      2016-01-18 07:40:00    2016-01-18 09.00.00    ATA     TU 32AIMC   
9      2016-01-18 07:45:00    2016-01-18 09.15.00    ATA     TU 31AIMK   

   Arrival delay  
0          260.0  
1           20.0  
2            0.0  
3            0.0  
4           22.0  
5           53.0  
6           10.0  
7           15.0  
8           16.0  
9           21.0

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 107833 entries, 0 to 107832
Data columns (total 9 columns):
 #   Column                    Non-Null Count   Dtype  
---  ------                    --------------   -----  
 0   Filght_date               107833 non-null  object 
 1   Flight_ID                 107833 non-null  object 
 2   Departure point           107833 non-null  object 
 3   Arrival point             107833 non-null  object 
 4   Scheduled_departure_time  107833 non-null  object 
 5   Scheduled_arrival_time    107833 non-null  object 
 6   STATUS                    107833 non-null  object 
 7   Aircraft_code             107833 non-null  object 
 8   Arrival delay             107833 non-null  float64
dtypes: float64(1), object(8)
memory usage: 7.4+ MB


In [6]:
df.describe()

Arrival delay
count  107833.000000
mean       48.733013
std       117.135562
min         0.000000
25%         0.000000
50%        14.000000
75%        43.000000
max      3451.000000

In [8]:
df.isna().sum()

Filght_date                 0
Flight_ID                   0
Departure point             0
Arrival point               0
Scheduled_departure_time    0
Scheduled_arrival_time      0
STATUS                      0
Aircraft_code               0
Arrival delay               0
dtype: int64

In [10]:
zero_count=(df==0).sum()
print(zero_count)

Filght_date                     0
Flight_ID                       0
Departure point                 0
Arrival point                   0
Scheduled_departure_time        0
Scheduled_arrival_time          0
STATUS                          0
Aircraft_code                   0
Arrival delay               38168
dtype: int64


In [9]:
profil=ProfileReport(df,title='Rapport Tunis_air')
profil.to_file('Rapport Tunis_air')

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

c:\Users\nabyp\OneDrive\Bureau\Analyse de données\PYTHON DATA\sklearn-env\Lib\site-packages\ydata_profiling\profile_report.py:379: UserWarning: Extension  not supported. For now we assume .html was intended. To remove this warning, please use .html or .json.
  warnings.warn(


Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

#   Observations

En ce basant sur le rapport nous pouvons constater que notre dataset contient 107833 enregistrements(lignes) et compte 9 variables (colonnes).On remarque plusieurs types de données :

2 variables de type DateTime

5 variables de type Texte  

1 variable Categorique 

1 variable numerique 

Aussi on se rend compte que notre dataset  occupe en memoire 7.4Mo , soit 72.0B par enregistrement.

On remarque cependant que notre dataset est propre c'est à dire qu'il n'y a pas de lignes en doubles encore moins de données manquantes.

Dans la colonne 'Arrival_delay' on remarque que le maximum 3451 est superieur au 95ieme percentile ce qui fait penser à une valeur aberrante

Un pourcentage élevé de zéros dans Arrival_delay suggère que cette variable pourrait avoir de nombreuses valeurs manquantes représentées par zéro. Il est crucial de déterminer comment gérer ces valeurs.

La matrice de correlation  montre la relation entre les deux variables "Retard à l'arrivée" et "STATUT".

La corrélation entre "Retard à l'arrivée" et lui-même est de 1, ce qui est attendu puisque toute variable est parfaitement corrélée à elle-même.

La corrélation entre "Retard à l'arrivée" et "STATUT" est de 0,03, indiquant une très faible relation positive entre ces deux variables.

De même, la corrélation entre "STATUT" et lui-même est de 1, et la corrélation entre "STATUT" et "Retard à l'arrivée" est également de 0,03.

En d'autres termes, les variables "Retard à l'arrivée" et "STATUT" sont presque indépendantes l'une de l'autre.